dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/MemoryAllocation.java

```diff
@@ -41,14 +68,22 @@ public void run(Program program) {
     }
 
     private void processAllocations(Program program) {
+        final ExpressionFactory expressions = ExpressionFactory.getInstance();
+        // FIXME: We should probably initialize depending on the allocation type of the alloc
```

Comment:
I don't get this comment. Isn't the check at line 82 checking this?

Reply:
By `allocation type` I mean the actual type of object that gets allocated. For mallocs, we allocate `bytes` (cause we don't know better) but for example `alloca` has a proper type. In that case, we could generate init events only at the correct offsets, e.g. every 4 bytes if allocating an `int32` array.